### PR TITLE
Realtime API voice assistant - initial spike

### PR DESCRIPTION
**Pull request by bcherry**
This is an initial spike to get voice-in/voice-out working with the model. It works!

### CODE REVIEW

File: agents/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "typescript": "^5.0.0"
   },
   "dependencies": {
-    "@livekit/rtc-node": "^0.7.0",
+    "@livekit/rtc-node": "file:../../node-sdks/packages/livekit-rtc",
```
Comment (bcherry) [inline agents/package.json]: currently depends on https://github.com/livekit/node-sdks/pull/264, we need to merge and update that before this one is ready

File: agents/src/audio.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { AudioFrame } from '@livekit/rtc-node';
```
Comment (nbsp) [inline agents/src/audio.ts:4]:
```suggestion
// SPDX-FileCopyrightText: 2024 LiveKit, Inc.
//
// SPDX-License-Identifier: Apache-2.0
import { AudioFrame } from '@livekit/rtc-node';
```

File: plugins/openai/src/voice_assistant/proto.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
+export enum Voice {
```
Comment (nbsp) [inline plugins/openai/src/voice_assistant/proto.ts:4]:
```suggestion
// SPDX-FileCopyrightText: 2024 LiveKit, Inc.
//
// SPDX-License-Identifier: Apache-2.0
export enum Voice {
```

Comment (bcherry) [reply plugins/openai/src/voice_assistant/proto.ts:4]: thanks - is there a way to automate this?

File: pnpm-lock.yaml
Comment (nbsp) [inline pnpm-lock.yaml:1]: this file shouldn't exist (probably a remnant of the manual linking to rtc-node)

File: plugins/openai/src/voice_assistant/index.ts [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+        ...command,
+        ...truncatedDataPartial,
+      });
+    }
+    this.ws.send(JSON.stringify(command));
+  }
+
+  private handleServerEvent(event: Record<string, unknown>): void {
+    const truncatedDataPartial = event['data']
+      ? { data: (event['data'] as string).slice(0, 30) + '…' }
```
Comment (nbsp) [inline plugins/openai/src/voice_assistant/index.ts:155]: i don't remember but there might have been a discussion about what the trunc character should be and whether it should be configurable. is this ellipsis the same as in python agents or do we use three dots there?

File: plugins/openai/src/voice_assistant/proto.ts
Comment (nbsp) [inline plugins/openai/src/voice_assistant/proto.ts:1]: let's make these enums `UPPER_SNAKE_CASE`? consistent with most other enums in the code

File: pnpm-lock.yaml
Comment (lukasIO) [reply pnpm-lock.yaml:1]: I think the file itself is needed?

Comment (nbsp) [reply pnpm-lock.yaml:1]: ah oops, thought this was a file on `agents/`

File: plugins/openai/src/voice_assistant/index.ts (hunk 4 above)
Comment (bcherry) [reply plugins/openai/src/voice_assistant/index.ts:155]: i don't think we have the same logging between them - we should bring them together but i was just adding it ad-hoc for now

**Pull request merged**